libsolidity/parsing/Parser.cpp

```diff
@@ -365,12 +365,12 @@ Parser::FunctionHeaderParserResult Parser::parseFunctionHeader(
 		Token::Value token = m_scanner->currentToken();
 		if (_allowModifiers && token == Token::Identifier)
 		{
-			// This can either be a modifier (function declaration) or the name of the
-			// variable (function type name plus variable).
-			if (
+			// If the name is empty, then this can either be a modifier (fallback function declaration)
+			// or the name of the state variable (function type name plus variable).
+			if (result.name->empty() && (
```

Comment:
I think this should be ``(result.name->empty() && !result.isConstructor)``, since the new-style constructors will also have an empty name.

Reply:
This is about identifiers that come after the parameter list, not before, so I think constructors do not apply here (apart from constructor calls of base classes).

Reply:
Ok, perhaps I'm wrong, let me think again :)

Reply:
Yes, you are perfectly right, thanks for spotting!